src/main/java/com/suse/saltstack/netapi/event/EventStream.java

```diff
@@ -129,99 +126,107 @@ public void removeEventListener(EventListener listener) {
 
     /**
-     * Closes the backing event stream and notifies all subscribed listeners that
-     * the event stream has been closed via {@link EventListener#eventStreamClosed()}.
+     * Closes the WebSocket {@link Session} and notifies all subscribed listeners.
+     * that the event stream has been closed via {@link EventListener#eventStreamClosed()}.
      * Upon exit from this method, all subscribed listeners will be removed.
      */
     public void close() {
+        // close the WebSocket session
+        if (!isEventStreamClosed()) {
+            try {
+                this.session.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        // notify all the listeners and cleanup
         synchronized (listeners) {
-            // notify all the listeners and cleanup
             for (EventListener listener : listeners) {
                 listener.eventStreamClosed();
             }
             // clear out the listeners
             listeners.clear();
-            // close the backing event stream
-            if (eventInput != null) {
-                eventInput.close();
-            }
-            // shut down the executor
-            executor.shutdownNow();
-            // reset processing flag
-            eventProcessingStarted = false;
         }
     }
 
     /**
-     * Helper method to determine whether the backing event stream is closed.
-     * @return Whether the {@link EventInput} stream is closed.
+     * On handshake completed, get the WebSocket Session and send
+     * a message to ServerEndpoint that WebSocket is ready.
+     * http://docs.saltstack.com/en/latest/ref/netapi/all/salt.netapi.rest_cherrypy.html#ws
+     *
+     * @param session The just started WebSocket {@link Session}.
+     * @param config The {@link EndpointConfig} containing the handshake informations.
+     * @throws IOException Exception thrown if something goes wrong sending message
+     * to the remote peer.
      */
-    public boolean isEventStreamClosed() {
-        return eventInput.isClosed();
+    @OnOpen
+    public void onOpen(Session session, EndpointConfig config) throws IOException {
+        this.session = session;
+        session.getBasicRemote().sendText("websocket client ready");
     }
 
     /**
-     * Helper method to determine whether event processing has started.
-     * @return Whether event processing has started.
+     * On each event received on the WebSocket,
+     * notify each listener with the event received.
+     *
+     * @param event The message received on this WebSocket.
      */
-    public boolean isEventProcessingStarted() {
-        return eventProcessingStarted;
+    @OnMessage
+    public void onMessage(String event) {
+        if (event != null && !event.equals("server received message")) {
+            synchronized (listeners) {
+                for (EventListener listener : listeners) {
+                    listener.notify(event.trim());
```

Comment:
@hustodemon Thank you for your suggestion, i'm changing the code according to Java 8.
@lucidd Yes, I <code>trim</code> the event string because it ends with <code>\n\n</code>